src/movies.js

```diff
@@ -1,15 +1,3123 @@
+      "duration": "2h 5min",
+      "genre": [
+        "Biography",
+        "Crime",
+        "Drama",
+        "Thriller"
+      ],
+      "rate": 8
+    },
+    {
+      "title": "Dead Poets Society",
+      "year": 1989,
+      "director": "Peter Weir",
+      "duration": "2h 8min",
+      "genre": [
+        "Comedy",
+        "Drama"
+      ],
+      "rate": 8
+    }
+  ];
+
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear(arr){
+  
+    const sortedByYearArr = [...arr].sort(function (firstItem, secondItem) {
+        if (firstItem.year == secondItem.year){
+            return firstItem.title.localeCompare(secondItem.title);
+        } 
+        
+        return firstItem.year - secondItem.year;
+    });
+
+    return sortedByYearArr;
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+
+function howManyMovies(arr){
+
+    let counter = 0;
+    const spielbergsFinest = arr.filter(function (movie){
+        if(movie.director == "Steven Spielberg" && movie.genre.includes("Drama")){
+            return counter++;
+        }
+        
+        return 0;
+    });
+  
+    return counter;
+  }
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(arr){   
+
+    const titlesOnly = arr.map(movie => movie.title);
+    const titlesRanked = [...titlesOnly].sort(function (firstItem, secondItem) {           
+        return firstItem.localeCompare(secondItem);
+    });
+    const shortArr = titlesRanked.slice(0, 20);
+
+    return shortArr;
+}
+
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(arr){
+
+      // edge cases
+      if(arr.length == 0){
+          return 0;
+      }
+
+    const rateArr = arr.map(movie => movie.rate);
+    const filterArr = rateArr.filter(rate => typeof rate == "number");
+    const sumRate = filterArr.reduce((acc, val) => acc + val);
+
+    return parseFloat((sumRate/arr.length).toFixed(2));
+}
+
 // Iteration 5: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr){
+
+    const dramas = arr.filter(movie => movie.genre.includes("Drama"));
+
+    return ratesAverage(dramas);
+}
+
 // Iteration 6: Time Format - Turn duration of the movies from hours to minutes
 
+function turnHoursToMinutes(arr){
+    
+    const updatedMovieDuration = arr.map(movie => {
+      let newDuration = 0;
+      let durationType = movie.duration;
+      const checkH = durationType.includes("h"); //returns true or false
+      const checkMin = durationType.includes("min"); //returns true or false
+    
+        if(checkH && checkMin){
+          let hMin = durationType.split(" ");
+          let newH = hMin[0].replace("h","");
+          let newMin = hMin[1].replace("min","");
+          newDuration = (Number(newH)*60) + (Number(newMin));
```

Comment:
You actually dont have to replace "h" and "min", parseInt does this far you:
parseInt("30 min") returns 30